spring-amqp/src/main/java/org/springframework/amqp/support/converter/SerializerMessageConverter.java

```diff
@@ -120,11 +141,40 @@ public Object fromMessage(Message message) throws MessageConversionException {
 		return content;
 	}
 
+	private Object deserialize(ByteArrayInputStream inputStream, Deserializer<Object> deserializer)
+			throws IOException {
+		ClassLoader classLoader = null;
+		try {
+			classLoader = (ClassLoader) new DirectFieldAccessor(deserializer).getPropertyValue("classLoader");
```

Comment:
Looks like we should do this only once, if we go the `DFA` way.
Or. Would be better do not pollute the code with the reflection at all and rely on our own `classload` and do `checkWhiteList()` if there was no `deserializer` injection.

It may be even better to provide `WhiteListDeserializer` impl instead...